internal/controller/notifier/notifier.go

```diff
@@ -47,8 +48,16 @@ func (watcher *Notifier) Register(ctx context.Context, worker string) (chan *rpc
 
 func (watcher *Notifier) Notify(ctx context.Context, worker string, msg *rpc.WatchInstruction) error {
 	slot, ok := watcher.workers.Load(worker)
-	if !ok {
-		return fmt.Errorf("%w: %s", ErrNoWorker, worker)
+
+	for !ok {
+		select {
+		case <-ctx.Done():
+			watcher.logger.Warnf("failed to notify watcher of worker %s due to timeout: %v", worker, ctx.Err())
+			return fmt.Errorf("%w: %s", ErrNoWorker, worker)
+		case <-time.After(time.Second):
+			watcher.logger.Infof("waiting for worker %s to re-connect...", worker)
+		}
+		slot, ok = watcher.workers.Load(worker)
```

Comment:
Perhaps we need to check this in a loop with some delay?

Otherwise a short flap will always result in a 30 second wait time, even through the worker could've been already registered after a couple of seconds.